lib/gh-linkify.js

```diff
@@ -0,0 +1,57 @@
+// github-style linkification for linkify-it
+//
+// Pass in a markdown-it parser instance and this will return a version fixed
+// up for the kind of linkification we're after.
+//
+// As far as I can determine, the basic algorithm is to linkify anything
+// starting with 'www.' and ending with a) a whitespace character, or b) a dot
+// followed by a whitespace character (in other words, include internal dots,
+// but exclude a trailing dot). There are plenty of cases where this results in
+// invalid links, but that's how they do it, so that's how we'll do it.
+//
+module.exports = function (parser) {
+  if (parser) {
+    var linkify = parser.linkify
+
+    linkify.set({fuzzyLink: false}) // turn off auto-linking normal hostnames
+    linkify.add('//', null) // turn off protocol-relative links
+
+    // linkify everything hostnamey starting with 'www.'
+    parser.linkify.add('www.', {
+      validate: function (text, pos, self) {
+        var tail = text.slice(pos)
+        var regex = /^\S*/
+        var match = tail.match(regex)[0]
+
+        // there's probably a nice regex way of excluding a trailing dot from
+        // these matches, but it's simple enough to just trim them off here
+        if (match.slice(-1) === '.') match = match.slice(0, -1)
```

Comment:
If anyone knows the regex fu that will allow internal dots in a linked host, but exclude a dot if it's the last thing that appears before any whitespace, we can fold that into the regex on line 23 and get rid of this line. I tried a few things but decided this was clearer (for me, at least).